### ISSUE TITLE
Docker plugin does not work correctly with Agent Images built using buildkit

### ISSUE THREAD

**Issue by dunespice** [jenkinsci/docker-plugin#867]
I believe the docker-plugin uses information from the docker agent image specifically
```
"ContainerConfig" {
  "WorkingDir": "/home/jenkins"
}
```
to work out where to install the remoting jar and where to log to when using the "Attach Docker container" connection method.
When agent images are built using buildkit this information is not present in ContainerConfig. It should be retrieved from Config instead.
I think this error is in getEffectiveRemoteFs() method of 
```
src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplate.java
```
The result is that the remoting can not start due to permission issues on the log files.

**Comment by xabolcs** [comment]
> When agent images are built using buildkit this information is not present in ContainerConfig. It should be retrieved from Config instead.

And that `final String containerWorkingDir = containerConfig == null ? null : containerConfig.getWorkingDir();` was originally introduced in commit f29ccbef37a4f081e084f06fb6c474a5869a9b01 as `remoteFs = image.getContainerConfig().getWorkingDir();`

**Comment by xabolcs** [comment]
This oneliner should fix.

```diff
diff --git a/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplate.java b/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplate.java
index f383a65..7ad0a1d 100644
--- a/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplate.java
+++ b/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplate.java
@@ -702,7 +702,7 @@ public class DockerTemplate implements Describable<DockerTemplate> {
         if (remoteFsOrNull != null) {
             return remoteFsOrNull;
         }
-        final ContainerConfig containerConfig = image.getContainerConfig();
+        final ContainerConfig containerConfig = image.getConfig();
         final String containerWorkingDir = containerConfig == null ? null : containerConfig.getWorkingDir();
         if (!StringUtils.isBlank(containerWorkingDir)) {
             return containerWorkingDir;
```

**Comment by basil** [comment]
Fixed in #942.
